assets/scripts/get-actions.js

```diff
@@ -1,12 +1,116 @@
+ * @param {string} key Came of the property in manifes.
+ * @param {object} propsAttributes Current atributes when this function executes.
+ *
+ * @since 1.0.5 Ucfirst path hotfix.
+ * @since 1.0.4 Moving to scripts folder.
+ * @since 1.0.0
+ */
+
+const multiplePropsActions = (setAttributes, attributes, key, propsAttributes) => {
+  const output = {};
+
+  // Set output as a object key with anonimus function callback.
+  for (const propType in attributes[key]) {
+
+    // Create functions for default values.
+    if (propType === 'default') {
+
+      for (const propName in attributes[key][propType]) {
+        if (attributes[key][propType].hasOwnProperty(propName)) {
+
+          output[`onChange${ucfirst(key)}${ucfirst(propName)}`] = function(value) {
+            setAttributes({
+              [key]: {
+                ...propsAttributes[key],
+                [propName]: value,
+              },
+            });
+          };
+
+        }
+      }
+    }
+  }
+
+  return output;
+};
+
+/**
+ * This method is for atributes with single property.
+ * This function generates callback for that propertiy value.
+ *
+ * @param {object} setAttributes Method for saving atributes.
+ * @param {string} key Came of the property in manifes.
+ *
+ * @since 1.0.5 Ucfirst path hotfix.
+ * @since 1.0.4 Moving to scripts folder.
+ * @since 1.0.0
+ */
+
+const siglePropsAction = (setAttributes, key) => {
+  const output = {};
+
+  // Set output as a object key with anonimus function callback.
+  // Keys first name must be uppercased.
+  output[`onChange${ucfirst(key)}`] = function(value) {
+    setAttributes({
+      [key]: value,
+    });
+  };
+
+  return output;
+};
+
+
+/**
+ * This method is for media atributes, it is property type `object` with default values of `id`, `url`, `title`.
+ * This function generates callback that saves `id`, `url` and `title` of attribute.
+ *
+ * @param {object} setAttributes Method for saving atributes.
+ * @param {string} key Came of the property in manifes.
+ *
+ * @since 1.0.5 Ucfirst path hotfix.
+ * @since 1.0.4 Moving to scripts folder.
+ * @since 1.0.0
+ */
+
+const mediaPropsAction = (setAttributes, key) => {
+  const output = {};
+
+  // Set output as a object key with anonimus function callback.
+  // Keys first name must be uppercased.
+  output[`onChange${ucfirst(key)}`] = function(value) {
+    setAttributes({
+      [key]: {
+        id: value.id,
+        url: value.url,
+        title: value.title,
+      },
+    });
+  };
+
+  return output;
+};
+
+
+
+/**
+ * Create attributes actions from blocks manifest.json.
  * Actions are passed in child components in order to update props on event.
  * Default function output is `onChange` + attribute name.
  * Example `onChangeContent`.
+ * Depending on prope
  *
- * @param {object} props Block props so we can get `setAttributes` method.
+ * @param {object} props Block props so we can get `setAttributes` Method.
```

Comment:
```suggestion
 * @param {object} props Block props so we can get `setAttributes` method.
```

No need to uppercase this.

Reply:
👍